internal/users/localentries/getpwent_c.go

```diff
@@ -0,0 +1,109 @@
+// Package localentries provides functions to access local passwd entries.
+//
+//nolint:dupl // The tests fail if we remove the duplicate C code.
+package localentries
+
+/* #include <stdlib.h>
+#include <pwd.h>
+#include <grp.h>
+#include <errno.h>
+#include <string.h>
+
+void unset_errno(void) {
+  errno = 0;
+}
+
+int get_errno(void) {
+  return errno;
+}
+*/
+//#cgo LDFLAGS: -Wl,--allow-multiple-definition
```

Comment:
Is this needed?

I guess you can just use `static` functions no?

Reply:
seems to work, thanks!